sys/kern/tty.c

```diff
@@ -876,57 +882,67 @@ static fileops_t tty_fileops = {
  * and the terminal has no associated session, make this terminal
  * the controlling terminal for the session. */
 static void maybe_assoc_ctty(proc_t *p, tty_t *tty) {
-  SCOPED_MTX_LOCK(all_proc_mtx);
+  assert(mtx_owned(all_proc_mtx));
+  assert(mtx_owned(&tty->t_lock));
 
   if (!proc_is_session_leader(p))
     return;
-
-  WITH_MTX_LOCK (&tty->t_lock) {
-    if (tty->t_session != NULL)
-      return;
-    session_t *s = p->p_pgrp->pg_session;
-    if (s->s_tty != NULL)
-      return;
-    tty->t_session = s;
-    s->s_tty = tty;
-    tty->t_pgrp = p->p_pgrp;
-  }
+  if (tty->t_session != NULL)
+    return;
+  session_t *s = p->p_pgrp->pg_session;
+  if (s->s_tty != NULL)
+    return;
+  tty->t_session = s;
+  s->s_tty = tty;
+  tty->t_pgrp = p->p_pgrp;
 }
 
-static int tty_vn_open(vnode_t *v, int mode, file_t *fp) {
+static int __tty_vn_open(vnode_t *v, int mode, file_t *fp, bool assoc) {
```

Comment:
Double underscore is usually used for functions and symbols provided by compiler and linker. Single underscore is ok though.

Reply:
I don't know, functions prefixed with double underscores are very common in Linux

Reply:
I changed it to a single underscore.